Explorer/Assets/DCL/MapRenderer/MapLayers/Pins/PinMarkerController.cs

```diff
@@ -0,0 +1,202 @@
+    {
+        internal delegate IPinMarker PinMarkerBuilder(
+            IObjectPool<PinMarkerObject> objectsPool,
+            IMapCullingController cullingController);
+
+        private readonly IObjectPool<PinMarkerObject> objectsPool;
+        private readonly PinMarkerBuilder builder;
+
+        public readonly Dictionary<Entity, IPinMarker> markers = new ();
+
+        private MapPinPlacementSystem system;
+        private MapPinTextureResolverSystem textureResolverSystem;
+        private MapPinDeletionSystem mapPinDeletionSystem;
+        private World world;
+
+        private bool isEnabled;
+
+        public PinMarkerController(
+            IObjectPool<PinMarkerObject> objectsPool,
+            PinMarkerBuilder builder,
+            Transform instantiationParent,
+            ICoordsUtils coordsUtils,
+            IMapCullingController cullingController)
+            : base(instantiationParent, coordsUtils, cullingController)
+        {
+            this.objectsPool = objectsPool;
+            this.builder = builder;
+        }
+
+        public async UniTask InitializeAsync(CancellationToken cancellationToken)
+        {
+        }
+
+        public void CreateSystems(ref ArchSystemsWorldBuilder<World> builder)
+        {
+            world = builder.World;
+            system = MapPinPlacementSystem.InjectToWorld(ref builder);
+            textureResolverSystem = MapPinTextureResolverSystem.InjectToWorld(ref builder);
+            mapPinDeletionSystem = MapPinDeletionSystem.InjectToWorld(ref builder);
+
+            system.SetQueryMethod(SetMapPinPlacementQuery);
+            textureResolverSystem.SetQueryMethod(SetMapPinTextureQuery);
+            mapPinDeletionSystem.SetQueryMethod(HandleEntityDestructionQuery);
+            system.Activate();
+            textureResolverSystem.Activate();
+            mapPinDeletionSystem.Activate();
+        }
+
+        [All(typeof(MapPinComponent))]
+        [Query]
+        private void SetMapPinPlacement(in Entity e, ref MapPinComponent mapPinComponent, ref PBMapPin pbMapPin)
+        {
+            if (!mapPinComponent.IsDirty)
+                return;
+
+            IPinMarker marker;
+            if (markers.TryGetValue(e, out IPinMarker pinMarker))
+            {
+                marker = pinMarker;
+            }
+            else
+            {
+                marker = builder(objectsPool, mapCullingController);
+                markers.Add(e, marker);
+            }
+
+            marker.SetPosition(coordsUtils.CoordsToPositionWithOffset(mapPinComponent.Position), mapPinComponent.Position);
+            marker.SetData(pbMapPin.Title, pbMapPin.Description);
+
+            if (isEnabled)
+                mapCullingController.StartTracking(marker, this);
+
+            mapPinComponent.IsDirty = false;
+        }
+
+        [All(typeof(MapPinComponent))]
+        [Query]
+        private void SetMapPinTexture(in Entity e, ref MapPinComponent mapPinComponent)
```

Comment:
`SetMapPinTexture` and `SetMapPinPlacement` can be invoked from the single query: thus, you will offload the ECS querying mechanism.